File: src/main/java/org/springframework/samples/petclinic/rest/mapper/PetTypeMapper.java
package org.springframework.samples.petclinic.rest.mapper;

import org.mapstruct.Mapper;
import org.springframework.samples.petclinic.owner.PetType;
import org.springframework.samples.petclinic.rest.dto.PetTypeDto;

@Mapper(componentModel = "spring")
public interface PetTypeMapper {

	PetTypeDto toDto(PetType petType);

}

File: src/main/kotlin/org/springframework/samples/petclinic/rest/mapper/PetTypeMapper.kt
package org.springframework.samples.petclinic.rest.mapper

import org.springframework.samples.petclinic.owner.PetType
import org.springframework.samples.petclinic.rest.dto.PetTypeDto
import org.springframework.stereotype.Component

@Component
class PetTypeMapper {

    fun toDto(petType: PetType): PetTypeDto {
        return PetTypeDto(
            id = petType.id,
            name = petType.name
        )
    }

}